classes/submission/form/PKPSubmissionSubmitStep3Form.inc.php

```diff
@@ -68,14 +68,21 @@ function fetch($request, $template = null, $display = false) {
 
 		$items = [];
 		$categoryDao = DAORegistry::getDAO('CategoryDAO'); /* @var $categoryDao CategoryDAO */
-		$categories = $categoryDao->getByContextId($context->getId());
-		if (!$categories->wasEmpty) {
-			while ($category = $categories->next()) {
-				$items[] = array(
-					'id' => $category->getId(),
-					'title' => $category->getLocalizedTitle(),
-				);
+		$result = $categoryDao->getByContextId($context->getId())->toArray();
```

Comment:
You can use `DAOResultFactory::toAssociativeArray` rather than `toArray`, which will return an array of `[context_id => Category]`. That'll save you looping through the categories below to find the parent category; you can just look it up by index.

Reply:
(apples to the other PRs as well)

Reply:
Thanks!